tests/Microsoft.Identity.Test.LabInfrastructure/LabServiceApi.cs

```diff
@@ -86,27 +108,16 @@ private LabResponse GetLabResponseFromAPI(UserQueryParameters query)
                 queryDict.Add("b2cProvider", "google");
             }
 
-            UriBuilder uriBuilder = new UriBuilder("http://api.msidlab.com/api/user");
-            uriBuilder.Query = string.Join("&", queryDict.Select(x => x.Key + "=" + x.Value.ToString()));
-
-            //Fetch user
-            string result = webClient.GetStringAsync(uriBuilder.ToString()).GetAwaiter().GetResult();
-
-            if (string.IsNullOrWhiteSpace(result))
+            if (query.UserContains != null)
             {
-                throw new LabUserNotFoundException(query, "No lab user with specified parameters exists");
+                queryDict.Add("usercontains", query.UserContains);
```

Comment:
What does this do?

Reply:
in order to get the msa user, the query needs to have "usercontains=MSIDLAB4_Outlook" to find the msa user

---
In reply to: [241020500](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/721#discussion_r241020500) [](ancestors = 241020500)

Reply:
I hope that going forward this will change in the more mature API, allowing us to simply request queryDict.Add("accountType", "MSA") or something similar

---
In reply to: [241241582](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/721#discussion_r241241582) [](ancestors = 241241582,241020500)